### PR TITLE
Various small fixes. [MDB IGNORE]

### LINKED ISSUE #2147: Grab feedback is very lackluster.
#### Description of issue
It is very unclear why various grab interactions don't work. Rather than showing an error they just show you nothing.

#### Difference between expected and actual behavior
Failing to upgrade or downgrade a grab, or failing to perform a grab action, should have some kind of feedback.

#### Steps to reproduce
- Grab person, stay on help intent.
- Try to upgrade grab.
- Nothing changes and no error is shown.

#### Client version, Server revision & Game ID
Current dev.

#### Issue bingo

- [X] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [X] Issue could be reproduced in multiple rounds
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### LINKED ISSUE #2166: Fountain of youth POI ignores species age ranges.
#### Description of issue
Fountain of youth uses hardcoded age ranges for the younger/older effect.

#### Difference between expected and actual behavior
Species with variant age ranges should have those ranges respected.

#### Steps to reproduce
Select neo-avian, click fountain of youth away site, note age doesn't change particularly.

#### Length of time in which bug has been known to occur
Probably since the age decl PR.

#### Client version, Server revision & Game ID
Client Version: 513
Server Revision: 0735ef59421f7ef49f89064870e0266c25a0423c - staging - 2021-10-17 (actually dev)
Game ID: cey-brMO
Current map: Tradeship Ocelot Alpha

#### Issue bingo

- [X] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [X] Issue could be reproduced in multiple rounds
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### LINKED ISSUE #2349: You can pour reagents directly into the chem master buffer, which causes issues.
#### Description of issue
Copying from downstream: https://github.com/ScavStation/ScavStation/issues/336
TLDR: You can pour chems into the buffer directly due to the chemmaster being an open container, which breaks.

#### Difference between expected and actual behavior
You should not be able to pour chems in without using a beaker that you insert.

#### Steps to reproduce
- Pick up bucket or bottle.
- Click on chemmaster.

#### Length of time in which bug has been known to occur
This was reported a very long time ago and I don't think anyone has looked at it.

#### Issue bingo
- [X] Issue could be reproduced at least once
- [X] Issue could be reproduced by different players
- [X] Issue could be reproduced in multiple rounds
- [ ] Issue happened in a recent (less than 7 days ago) round
- [X] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### LINKED ISSUE #2425: Pimpin' ride won't move without keys D:
#### Description of issue

You can't move janicart without keys in your hands, even when you try to grab-move it.
Also I suggest, to add ability to leave keys in the Ignition switch and move based on presence of it.

#### Difference between expected and actual behavior

I suppose you must be able to move it with your hands

#### Steps to reproduce

1) Get /obj/structure/janitorialcart
2) Try to grab it and than move 
3) Get funny warning message about need to held keys

#### Length of time in which bug has been known to occur

1 day

#### Client version, Server revision & Game ID

Client Version: 514
Server Revision: 68216f3b20a02b81ceff28fc282be4a259ab398f - ceff28fc282be4a259ab398f - 2022-05-22
Game ID: ch3-de9r

#### Issue bingo

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### LINKED ISSUE #2438: Persistence didnt track noticeboards correctly
#### Description of issue
Persistence didnt track noticeboards correctly
it resets direction after 1 restart
изображение
and noticeboard dissapears after 2 restarts
изображение

sometimes it vanished afred 1 restart with all papers, sometimes leave "persistent" paper that exists until it is destroyed
#### Difference between expected and actual behavior
I think it should track it until someone or something destroy it

#### Steps to reproduce
1) craft noticeboard
2) pin some papers
3) dont touch noticeboard and restart server 2 times
#### Client version, Server revision & Game ID

Client Version: 514
Server Revision: c2377f69454640262b020b26407fc27de8786968 - dev - 2022-05-27
Game ID: ch8-aBbe

#### Issue bingo

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### LINKED ISSUE #2439: noticeboards direction
#### Description of issue
noticeboards direction shifted 180 degrees if you craft it

Скриншот 27-05-2022 054553
Скриншот 27-05-2022 054547

#### Difference between expected and actual behavior
it should be faced to you after crafting

#### Steps to reproduce
1) stay face to wall
2) craft noticeboard
3) direction is incorrect

#### Client version, Server revision & Game ID

Client Version: 514
Server Revision: c2377f69454640262b020b26407fc27de8786968 - dev - 2022-05-27
Game ID: ch8-aBbe

#### Issue bingo

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### LINKED ISSUE #2442: cant close someone mouth
#### Description of issue
if you try to close someone mouth or just grab it, you will fail
изображение

#### Difference between expected and actual behavior
you should grab the target and close their mouth

#### Steps to reproduce
1) select mouth 
2) grab someone

#### Client version, Server revision & Game ID

Client Version: 514
Server Revision: c2377f69454640262b020b26407fc27de8786968 - dev - 2022-05-27
Game ID: ch8-aBbe

#### Issue bingo

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### PR DESCRIPTION
**Pull request by MistakeNot4892**
- Fixes #2349.
- 'Fixes' #2328.
- Fixes #2166.
- Fixes #2147.
- Fixes #2442.
- Fixes #2439.
- Fixes #2438.
- 'Fixes' #2084.
- Fixes #2425.

### CODE REVIEW

**Review by afterthought2 — DISMISSED** [review]

File: code/controllers/subsystems/initialization/materials.dm [hunk 1]
```diff
@@ -146,6 +146,21 @@ SUBSYSTEM_DEF(materials)
...
+		return
+	mat_name = lowertext(mat_name)
+	if(isnull(materials_by_name[mat_name]))
+		var/list/mats = decls_repository.get_decls_of_subtype(/decl/material)
+		for(var/mat in mats)
+			var/decl/material/mat_decl = mats[mat]
+			if(lowertext(mat_decl.name) == mat_name)
+				materials_by_name[mat_name] = mat_decl
+				return mat_decl
+		materials_by_name[mat_name] = FALSE
```
Comment (afterthought2) [inline code/controllers/subsystems/initialization/materials.dm]: Why not just create this list once (either at init or the first time this proc is called), putting in all the decls? This is slower and also has a questionable return type.

File: code/controllers/subsystems/initialization/materials.dm [hunk 2]
```diff
@@ -146,6 +146,21 @@ SUBSYSTEM_DEF(materials)
 		global.default_strata_type_by_z[s_key] = pick(subtypesof(/decl/strata))
 	return global.default_strata_type_by_z[s_key]
 
+/datum/controller/subsystem/materials/proc/get_material_by_name(var/mat_name)
+	RETURN_TYPE(/decl/material)
```
Comment (afterthought2) [inline code/controllers/subsystems/initialization/materials.dm]: What does this do here? Clearly this proc can return a decl, null, or 0 just by inspecting the code, and yet we are not getting a compile-time error.

Comment (MistakeNot4892) [reply code/controllers/subsystems/initialization/materials.dm]: Linter told me to, if I'm being honest.

File: code/controllers/subsystems/initialization/materials.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/controllers/subsystems/initialization/materials.dm]: This was a partial commit I meant to snip out later when I realized materials_by_name existed but was incorrectly populated. Gone now.

File: code/game/atoms_movable.dm [hunk 3]
```diff
@@ -202,19 +202,19 @@
 		if(isturf(loc))
 			buckled_mob.glide_size = glide_size // Setting loc apparently does animate with glide size.
 			buckled_mob.forceMove(loc)
-			refresh_buckled_mob()
+			refresh_buckled_mob(0)
```
Comment (MistakeNot4892) [inline code/game/atoms_movable.dm:205]: This is a side-change that prevents atoms that a mob is buckled to from causing them to slide around via animate() when changing dir, specifically due to the janicart looking like absolute bollocks when moving.

**Review by afterthought2 — DISMISSED** [review]

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**